security/advancedtls/advancedtls.go

```diff
@@ -475,3 +478,28 @@ func cloneTLSConfig(cfg *tls.Config) *tls.Config {
 	}
 	return cfg.Clone()
 }
+
+// GetCertificateWithSNI is a helper function that returns the certificate
+// that matches the SNI field based on the given ClientHelloInfo
+// if GetCertificate returns a list of certificates
+func (o *ServerOptions) getCertificateWithSNI(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	certificates, err := o.GetCertificate(clientHello)
+	if err != nil {
+		return nil, err
+	}
+	if len(*certificates) == 0 {
```

Comment:
Is this the right way to check the length of a slice?

Reply:
Let me double check on this :)

Reply:
I changed this to `len(certificates)` because now certificates is a slice of ptr to certificates.